Ignore secrets yaml on command line

cvwillegen opened the pull request:
# What does this implement/fix? 

When ``esphome`` is run from the CLI to update a subset of the configuration files, or all of them, the ``secrets.yaml`` file - when specified - yields an error. This makes it hard to do, for instance, ``esphome upload *yaml``. This code makes sure to skip the files ``secrets.yaml`` and ``secrets.yml`` when specified on the command line.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [X] ESP8266

## Example entry for `config.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example config.yaml

```

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

cvwillegen committed:
Add local MAC address to WiFi info

When connecting to an AP, it is sometimes necessary to know the local MAC of the ESP device to add it to the network. This info is now printed in the log.

cvwillegen committed:
Fixes for clang-format

cvwillegen committed:
Use get_mac_address_pretty()

Formatting the MAC manually is not as pretty...

cvwillegen committed:
Merge pull request #1 from cvwillegen/cvwillegen-add-mac-to-wifi-info

Add local MAC address to WiFi info

cvwillegen committed:
Merge branch 'esphome:dev' into dev

cvwillegen committed:
Merge branch 'esphome:dev' into dev

cvwillegen committed:
Merge branch 'esphome:dev' into dev

cvwillegen committed:
Merge branch 'esphome:dev' into dev

cvwillegen committed:
Merge branch 'esphome:dev' into dev

cvwillegen committed:
Merge branch 'esphome:dev' into dev

cvwillegen committed:
Add names of secrets files to constants

When ``esphome`` is run from the CLI to update a subset of the configuration files, or all of them, the ``secrets.yaml`` file - when specified - yields an error. This makes it hard to do, for instance, ``esphome upload *yaml``. This makes sure to skip the files ``secrets.yaml`` and ``secrets.yml`` when specified on the command line.

cvwillegen committed:
When secrets files are specified, skip them

cvwillegen committed:
Use lazy % formatting in logging functions

Found by pylint during CI run.

cvwillegen committed:
Check if args contain configs before checking them

cvwillegen committed:
Reformatted using black

oxan commented on esphome/__main__.py:
```diff
@@ -781,6 +777,20 @@ def run_esphome(argv):
+        for secret in CONF_SECRETS:
+            if secret.casefold() in (
+                config.casefold() for config in args.configuration
+            ):
```
This check is too strict, we only want to exclude the literal filename `secrets.yaml`, not any path that contains that.

cvwillegen replied:
Used your example code to skip files, thanks!

oxan commented on esphome/__main__.py:
```diff
@@ -781,6 +777,20 @@ def run_esphome(argv):
+            ):
+                _LOGGER.warning(
+                    "%s was specified on the command line. Skipping...", secret
+                )
```
This is not very descriptive, might be nice to add a note why this file is being skipped.

oxan commented on esphome/__main__.py:
```diff
@@ -781,6 +777,20 @@ def run_esphome(argv):
+                    for conf in args.configuration
+                    if conf.casefold() != secret.casefold()
+                ]
+
```
It's probably easier to do this in the for loop below, something like this:
```python
    for conf_path in args.configuration:
      if any(basename(conf_path) is x for x in CONF_SECRETS):
        _LOGGER.warning("skipping secrets file")
        continue
```
This also has the advantage that the `hasattr(args, "configuration")` is not necessary.

oxan commented on esphome/const.py:
```diff
@@ -591,6 +591,7 @@
 CONF_SDO_PIN = "sdo_pin"
 CONF_SECOND = "second"
 CONF_SECONDS = "seconds"
+CONF_SECRETS = ["secrets.yaml", "secrets.yml"]
```
This shouldn't be a `CONF_` constant, as those are for keys in the configuration YAML.

cvwillegen committed:
Specify names of secrets files

Secrets files names should be specified outside of config key names.

cvwillegen committed:
Skip secrets files in a more concise way

The secrets files were skipped at the wrong time of processing. Move the check down a bit. Also provide a better warning when skipping a secrets file.

cvwillegen committed:
Remove erroneous whitespace.

oxan commented:
@jesserockz I think this is a good idea for usability, but it's technically a breaking change as you can no longer have a config named `secrets.yaml`. I don't think that's much of a problem, but thoughts?

jesserockz reviewed (approved):
That's fine. I would not expect people to have that anyway.

oxan commented:
Thank you for your contribution!

cvwillegen commented:
This has been an itch to scratch, but you're welcome! My second addition to esphome, both personal itches, but probably useful in the wild!
